### PR TITLE
Msde collect and download package command

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47615

## Description
add a command that combines 4 commands to request and download an investigation package.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Microsoft Defender for Endpoint",
     "description": "Microsoft Defender for Endpoint (previously Microsoft Defender Advanced Threat Protection (ATP)) is a unified platform for preventative protection, post-breach detection, automated investigation, and response.",
     "support": "xsoar",
-    "currentVersion": "1.7.0",
+    "currentVersion": "1.7.1",
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/pack_metadata.json:5]: bump version by minor

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 2]
```diff
@@ -143,6 +143,7 @@ After you successfully execute a command, a DBot message appears in the War Room
 46. endpoint
 47. microsoft-atp-indicator-batch-update
 48. microsoft-atp-get-alert-by-id
+49. microsoft-atp-request-and-download-investigation-package
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:146]: add the full command section too

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 3]
```diff
@@ -5678,6 +5678,54 @@ script:
     - contextPath: MicrosoftATP.MachineAlerts.MachineID
       description: The alerts machine ID.
       type: String
+  - arguments:
+    - default: false
+      description: The machine ID.
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:5683]: From where can the user get this ID? Another command output?
Add to the description.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 4]
```diff
@@ -5678,6 +5678,54 @@ script:
...
+      isArray: false
+      name: machine_id
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:5685]: same here

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 5]
```diff
@@ -2700,6 +2700,35 @@ def get_machine_action_by_id_command(client: MsClient, args: dict):
...
+    md_results = {
+        'Machine Action Id': res.get('id'),
+        'MachineId': res.get('machineId'),
+        'Status': res.get('status'),
+    }
+    return [fileResult('Response Result.gz', f_data.content),
+            CommandResults(
+                outputs_prefix='MicrosoftATP.InvestigationUR',
+                outputs=res,
+                readable_output=tableToMarkdown('Machine Action:', md_results, is_auto_json_transform=True)
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2728]: Why do you need `is_auto_json_transform`? Which field is complex?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 6]
```diff
@@ -4439,13 +4468,12 @@ def run_polling_command(client: MsClient, args: dict, cmd: str, action_func: Cal
...
     # distinguish between the initial run, which is the upload run, and the results run
     is_first_run = 'machine_action_id' not in args
-    demisto.debug(f'polling args: {args}')
     if is_first_run:
         command_results = action_func(client, args)
         outputs = command_results.outputs
         # schedule next poll
         polling_args = {
-            'machine_action_id': outputs.get('action_id'),
+            'machine_action_id': outputs.get('action_id') or command_results.raw_response.get("id"),
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:4476]: Isn't the output extracted from the raw_response? Why would the second work if the first is empty?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 7]
```diff
@@ -4459,14 +4487,20 @@ def run_polling_command(client: MsClient, args: dict, cmd: str, action_func: Cal
         return command_results
 
     # not a first run
-
     command_result = results_function(client, args)
     action_status = command_result.outputs.get("status")
-    command_status = command_result.outputs.get("commands", [{}])[0].get("commandStatus")
+    demisto.debug(f"action status is: {action_status}")
+    if len(command_result.outputs.get("commands", [])) > 0:
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:4493]:
```suggestion
    if command_result.outputs.get("commands", []):
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 8]
```diff
@@ -4459,14 +4487,20 @@ def run_polling_command(client: MsClient, args: dict, cmd: str, action_func: Cal
...
+        command_status = command_result.outputs.get("commands", [{}])[0].get("commandStatus")
+    else:
+        command_status = 'Completed' if action_status == "Succeeded" else None
     if action_status in ['Failed', 'Cancelled'] or command_status == 'Failed':
-        raise Exception(
-            f'Command {action_status}. Additional info: {command_result.outputs.get("commands", [{}])[0].get("errors")}')
+        error_msg = f"Command {action_status}."
+        if len(command_result.outputs.get("commands", [])) > 0:
```
Comment (yaakovi) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:4499]:
```suggestion
        if command_result.outputs.get("commands", []):
```

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**